### PR TITLE
CXXCBC-310 LostTxnCleanup wasn't stopping right away

### PR DESCRIPTION
**Pull request by davidkelly**
When shutting down, for instance, the lost attempt loop would not stop sometimes, leading to issues with tests sometimes.  Now, I'm only accessing the running_ bool under the lock, so it is safe to use it in the wait calls as well.  Seems to have been an issue, simple fix.

Also - I was creating a client record doc if it was missing, when deleting, which does nothing.

### CODE REVIEW

File: core/transactions/transactions_cleanup.cxx [hunk 1]
```diff
@@ -526,7 +517,7 @@ transactions_cleanup::attempts_loop()
 void
 transactions_cleanup::add_attempt(attempt_context& ctx)
 {
-    auto& ctx_impl = static_cast<attempt_context_impl&>(ctx);
+    auto& ctx_impl = dynamic_cast<attempt_context_impl&>(ctx);
```
Comment (avsej) [inline core/transactions/transactions_cleanup.cxx:520]: What is the reason if this change? I think it is exactly one kind of derived class here, right? I wonder if we really need this dynamic_cast, which can do extra checks in runtime and might return `nullptr`.

https://en.cppreference.com/w/cpp/language/dynamic_cast
> If the cast is successful, dynamic_cast returns a value of type new-type. If the cast fails and new-type is a pointer type, it returns a null pointer of that type. If the cast fails and new-type is a reference type, it throws an exception that matches a handler of type [std::bad_cast](https://en.cppreference.com/w/cpp/types/bad_cast).

File: core/transactions/transactions_cleanup.cxx [hunk 2]
```diff
@@ -273,135 +274,127 @@ transactions_cleanup::create_client_record(const couchbase::transactions::transa
...
+                    for (const auto& [override, param] : value.get_object()) {
+                        if (override == FIELD_OVERRIDE_ENABLED) {
+                            details.override_enabled = param.get_boolean();
+                        } else if (override == FIELD_OVERRIDE_EXPIRES) {
+                            details.override_expires = param.as<std::uint64_t>();
+                        }
+                    }
+                } else if (key == FIELD_CLIENTS_ONLY) {
+                    for (const auto& [other_client_uuid, cl] : value.get_object()) {
+                        uint64_t heartbeat_ms = parse_mutation_cas(cl.at(FIELD_HEARTBEAT).get_string());
```
Comment (avsej) [inline core/transactions/transactions_cleanup.cxx:318]: could you use the namespace for `uint64_t` and `int64_t` below to ensure that stdlib versions will be used

Comment (davidkelly) [reply core/transactions/transactions_cleanup.cxx:318]: ooh - good catch.   I'll sweep through and make it consistent.

File: core/transactions/transactions_cleanup.cxx (hunk 1 above)
Comment (davidkelly) [reply core/transactions/transactions_cleanup.cxx:520]: Oh -- IIRC CLion had suggested it.  I never check the result anyways, as this has to succeed given the class hierarchy.   Ya back to static_cast.

**Review by avsej — APPROVED** [review]

**Pull request merged**